src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp

```diff
@@ -370,7 +369,7 @@ void QMCCostFunctionBatched::checkConfigurations(EngineHandle& handle)
           RecordsOnNode[is][LOGPSI_FIXED] = opt_data.get_log_psi_fixed()[ib];
           RecordsOnNode[is][LOGPSI_FREE]  = opt_data.get_log_psi_opt()[ib];
 
-          handle.takeSample(energy_list, dlogpsi_array, dhpsioverpsi_array, ib);
+          handle.takeSample(energy_list, dlogpsi_array, dhpsioverpsi_array, ib, is);
```

Comment:
Could you adjust the takeSample API and move the call out of the ib loop?
takeSample(energy_list, dlogpsi_array, dhpsioverpsi_array, base_sample_index);
In the takeSample implementation, handle the ib loop.

Reply:
Ok, I've put the ib loop inside takeSample.